uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/pageObjects/Page.java

```diff
@@ -11,16 +13,29 @@ public class Page {
 
     public Page(WebDriver driver) {
         this.driver = driver;
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
```

Comment:
🤔 

This looks like a 5-second wait on every page creation.

Reply:
This changes the default timeout of 0 (fail immediately when looking for an element if the page isn't completely loaded already) to a 5 second timeout. It does not introduce an arbitrary wait in the case of the element being found right away. 

This is an attempt to make the tests more robust and it should not have a significant impact on the execution time. It's most likely to affect the check in the page object constructors that validates that we're already on the expected page.

Reply:
Got it. I wish this was clearly explained in the code, for everybody's knowledge. 

Personally, I would have made this change in its own commit, so that I could have added what you just wrote in the description.

Another idea is to abstract this code into a function with a name of the type "ensure that we wait long enough, to the limit of 5 seconds, for html elements to be populated, instead of immediately failing".

Though waiting for these things is generally considered an anti-pattern. We should really wait for the page to be fully loaded before starting testing on it.